storage/mysql/log_storage.go

```diff
@@ -265,6 +266,12 @@ func (t *logTX) QueueLeaves(leaves []trillian.LogLeaf) error {
 		if len(leaf.LeafHash) != t.ts.hashSizeBytes {
 			return fmt.Errorf("Queued leaf must have a hash of length %d", t.ts.hashSizeBytes)
 		}
+
+		// Validate the hash as a consistency check that the data was received OK. Note: at
+		// this stage it is not a merkle tree hash for the leaf
```

Comment:
Possibly. I retained the distinction so we still have a usable hash of what the client sent until it's been integrated. It could also potentially be removed to avoid the confusion a different way because if we decide it's bad when we come to integrate we've already told the client we accepted it.